lib/cadence/workflow/poller.rb

```diff
@@ -68,17 +69,19 @@ def poll_loop
       end
 
       def poll_for_task
-        client.poll_for_decision_task(domain: domain, task_list: task_list)
+        connection.poll_for_decision_task(domain: domain, task_list: task_list)
       rescue StandardError => error
         Cadence.logger.error("Unable to poll for a decision task: #{error.inspect}")
         nil
       end
 
       def process(task)
-        client = Cadence::Client.generate
+        connection = Cadence::Connection.generate(config.for_connection)
```

Comment:
could/should this be done inside of `DecisionTaskProcessor.initialize`?

Reply:
We can do this, yes. The benefit of this approach is dependency injection, but I think this one makes sense to move into the `DecisionTaskProcessor`. I'll fix it, thanks